### PR TITLE
refactor: upgrade class-transformer to newest version

### PR DESCRIPTION
**Pull request by kaufmo**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +45,42 @@ All notable changes to this project will be documented in this file. This projec
     intl.formatMessage(messages.saveUnsavedChanges);
     ```
 
+-   class-transformer changes:
```
Comment (johnnyomair) [inline CHANGELOG.md:48]: Do we need to document class-transformer breaking changes? Aren't they documented in their CHANGELOG already?

Comment (kaufmo) [reply CHANGELOG.md:48]: only ClassConstructor is not documented. I documented these changes here for easier comet migrations i can remove if you want?

Comment (nsams) [reply CHANGELOG.md:48]: imho it is a good a idea to give a short overview here.

Comment (nsams) [reply CHANGELOG.md:48]: (this is not an "Incompatible Change" though, as we didn't have a stable release so far)

**Review by johnnyomair — DISMISSED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (johnnyomair) [reply CHANGELOG.md:48]: Okay, I'd leave it as it is. This changelog is good for our developers switching from one canary version to the next. We should remove it before the stable release though

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Pull request merged**